pyriemann/transferlearning_meta.py

```diff
@@ -0,0 +1,122 @@
+
+import numpy as np
+from sklearn.model_selection import ShuffleSplit
+from sklearn.base import BaseEstimator, TransformerMixin, ClassifierMixin
+from .preprocessing import Whitening
+
+
+class TLSplitter():
+    def __init__(self, target_train_frac=0.80, n_splits=5):
+        self.target_train_frac = target_train_frac
+        self.n_splits = n_splits
+
+    def split(self, X, y, meta):
+
+        # index of all source data points
+        idx_source = np.where(~meta['target'])[0]
+
+        # index of training-split for the target data points
+        ss_target = ShuffleSplit(
+            n_splits=self.n_splits,
+            train_size=self.target_train_frac).split(X[meta['target']])
+        for train_idx_target, test_idx_target in ss_target:
+            train_idx_target = list(
+                meta[meta['target']].index[train_idx_target])
+            train_idx = np.concatenate([idx_source, train_idx_target])
+            test_idx_target = list(meta[meta['target']].index[test_idx_target])
+            test_idx = test_idx_target
+            yield train_idx, test_idx
+
+    def get_n_splits(self, X, y, meta):
+        return self.n_splits
+
+
+class DCT(BaseEstimator, TransformerMixin):
+    '''
+    No transformation of the data points between the domains.
+    This is what we call the direct (DCT) method.
+    '''
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y, meta):
+        return self
+
+    def transform(self, X, y=None, meta=None):
+        return X
+
+    def fit_transform(self, X, y, meta):
+        return self.fit(X, y, meta).transform(X, y, meta)
+
+
+class RCT(BaseEstimator, TransformerMixin):
+    '''
+    Re-center (RCT) the data points from each domain to the Identity.
+    '''
+
+    def __init__(self):
+        pass
+
+    def fit(self, X, y, meta):
+        self.whitening_ = {}
```

Comment:
I think we can leave the `meta` alternative as is for now, since we're not evolving it at the moment...

Also, I am having 2nd thoughts about calling `whitening` the function/attribute related to the re-centering. Yeah, of course, the operation corresponds to a whitening and it is clear for someone coming from a multivariate signal processing or machine learning background. However, for people with a Riemannian geometry flavor (which are suppose to be our target public too) this operation is a re-centering to the Identity (or a parallel transport to be more rigorous) and the word `whitening` does not mean anything in particular... Since RPA is based on geometric transformations of data points in a manifold, I would be more inclined to leave the names of its operations as close as possible to geometric ones and not their signal processing counterparts.

Reply:
Why not call the private attribute `_recenter` but be sure to use the `Whitening` step? Another option would be to create a `ParallelTransport` class that sends points from one place to another in the SPD manifold. The `Whitening` and `ReCentering` would simply be methods that take points to the Identity, but other works like this one [1](https://arxiv.org/pdf/1807.10479.pdf) would send the dataset elsewhere. Wdyt ?

Reply:
Ok for me to use `_recenter` as private attribute.

> Another option would be to create a ParallelTransport class that sends points from one place to another in the SPD manifold. The Whitening and ReCentering would simply be methods that take points to the Identity, but other works like this one [1](https://arxiv.org/pdf/1807.10479.pdf) would send the dataset elsewhere. Wdyt ?

Yes, great idea, but in another PR I suggest.

Btw, do you know function `transport`, never added in the API?
https://github.com/pyRiemann/pyRiemann/blob/master/pyriemann/utils/tangentspace.py#L375
It could be great to develop the parallel transport functions / classes of pyRiemann. You seem the good guy to do that ;-)